frontend/src/router/index.ts

```diff
@@ -140,15 +140,22 @@ export default function getRouter() {
       });
     }
 
+
+    // Uploading navigation guard
+    if (appStore.getIsUploading) {
+      if (!confirm('Navigation may cancel upload(s) in progress. ' +
+        'Please confirm you want to navigate from current page.')) {
+        return false;
+      }
+    }
+
```

Comment:
I'm a bit unsure of order of operations and lifecycle here. Should this be in the `beforeRouteLeave` stage of the lifecycle instead?